app/controllers/submissions_controller.rb

```diff
@@ -124,7 +124,14 @@ def populate_file_manager
       end
       entries = get_all_file_data(revision, grouping, '')
     end
-    render json: entries
+
+    response = {
+      entries: entries,
+      only_required_files: assignment.only_required_files,
+      required_files: assignment.assignment_files.collect(&:filename)
```

Comment:
This is better but can still be improved: use `pluck`, which will access the fields directly from the database. (`collect` will first instantiate the `AssignmentFile` objects and then grab their `filename` attributes, which is slower)

Also please call `sort` on the result to make sure the filenames are sorted (and test this out by creating required files with names that are out of order alphabetically)